src/interfaces/form/form.interface.ts

```diff
@@ -9,34 +9,50 @@ export interface InputComponentProps {
   inputHandler: ChangeEventHandler<HTMLInputElement>;
 }
 
-export interface FormProps {
-  [key: string]: any;
+export interface LoginFormFields {
+  email: string;
+  password: string;
+}
+export interface SignupFormFields {
+  username: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+}
+
+export type SignupFormFieldName = keyof SignupFormFields;
+
+export type LoginFormFieldName = keyof LoginFormFields;
+
+export type FormFields = SignupFormFields | LoginFormFields;
+
+export type FormFieldName = keyof FormFields;
+
+export interface FormProps<T> {
+  values: T;
   validationRules: ValidationProps;
 }
 
 export type Validator = (
   inputName: string,
-  values: FormProps,
+  values: FormFields,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   arg?: any,
 ) => string;
 
 export interface FormValidation {
   validator: Validator;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   arg?: any;
 }
 
 export interface ValidationProps {
   [key: string]: FormValidation | FormValidation[];
 }
 
-export interface FormUIProps extends FormProps {
```

Comment:
removed from here (since it's component specific and not really related to forms in that sense), and modified bit